auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -2074,6 +2074,7 @@ SecurityIdentity getSourceIdentity() {
         boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableException {
             // At this stage, we just verify that the evidence principal matches, and verify it with the realm.
             final Principal evidencePrincipal = evidence.getDecodedPrincipal();
+            if (evidencePrincipal == null) { log.tracef("Authentication Failed"); }
```

Comment:
I think this can be removed. At this point, we haven't called `verifyEvidence` yet so we don't know whether authentication has failed here.